pkg/config/expand.go

```diff
@@ -737,147 +529,53 @@ func (ref modReference) validate(bp Blueprint) error {
 	return nil
 }
 
-// this function validates every field within a varReference struct and that
-// the reference must be to the same or earlier group.
-// ref.GroupID: this group must exist or be the value "deployment"
-// ref.ID: must be an existing module ID or "vars" (if groupID is "deployment")
-// ref.name: must match a module output name or deployment variable name
-// ref.explicitInterGroup: intergroup references must explicitly identify the
-// target group ID and intragroup references cannot have an incorrect explicit
-// group ID
-func (ref varReference) validate(bp Blueprint) error {
+// validateModuleSettingReference checks that a reference within module settings is valid
```

Comment:
I think this comment is tautological. I would like to see a few big failures clearly stated.

1. References to deployment variables must exist
2. References to the outputs of other modules must exist (either intra- or inter- group)
  * Packer is a sub-case of this and doesn't really need to be called out in the comment 
4. Intergroup references must be in the right order

If I'm missing more, please add them.